File: sum-root-to-leaf-numbers.cpp
//Runtime: 3 ms
class Solution {
public:
    int ans = 0;

    void solve(TreeNode* root, string t)
    {
        if(!root->left && !root->right)
            ans+=stoi(t);

        if(root->left)
            solve(root->left, t+to_string(root->left->val));

        if(root->right)
            solve(root->right, t+to_string(root->right->val));
    }

    int sumNumbers(TreeNode* root) {
        if(!root)
            return 0;
        solve(root,to_string(root->val));
        return ans;
    }
};

File: sum-root-to-leaf-numbers.py
# Runtime: 48 ms


class Solution(object):
    ans = 0

    def solve(self, root, s):
        if not root.left and not root.right:
            self.ans += int(s)
            return
        if root.left:
            self.solve(root.left, s + str(root.left.val))
        if root.right:
            self.solve(root.right, s + str(root.right.val))

    def sumNumbers(self, root):
        if root:
            self.solve(root, str(root.val))
        return self.ans
